design/ceph/ceph-orchestrator-osd-add-remove.md

```diff
@@ -0,0 +1,169 @@
+# Adding and removing Ceph osds via Ceph orchestrator interface
+
+## Goals
+
+1. Give Ceph's Rook Orchestrator Plugin the ability to add and remove OSDs to/from a Rook-Ceph
+   cluster running in Kubernetes.
+2. Users of the Ceph Orchestrator CLI and/or Ceph Dashboard should be able to
+   create OSDs on nodes with as much complexity as Ceph is able.
+3. Do not introduce unnecessary complexity into Rook or into Rook's CephCluster CRD.
+
+## Proposal
+
+### Removing OSDs
+Removing OSDs is the simplest place to start with this design. When removing an OSD or OSDs via the
+Rook Orchestrator Module, the Ceph mgr running in Kubernetes will run a Kubernetes Job that will run
+the required `ceph-volume` command necessary to remove the OSD(s) from a node as desired by the Ceph
```

Comment:
I don't think we'll need a ceph-volume command, but I certainly see this as the place to automate [these instructions](https://rook.io/docs/rook/v1.2/ceph-osd-mgmt.html#remove-an-osd). The missing piece for automating those instructions has been that user confirmation to remove the osd, which the mgr module is driven by.

Reply:
If we need a c-v command this would likely be a zap subcommand.

Reply:
WIP around OSd removal in the Ceph Orchestrator Group: [mgr/osd_support: new module for osd utility](https://github.com/ceph/ceph/pull/32677).
The aim of this PR is to provide a new orchestrator module that can be used by others modules to resolve the problem of remove OSD in a safe way through this command

`$ ceph osd drain $osd_ids ([$osd_ids])`

I think we should include all the manual [required steps ](https://rook.io/docs/rook/v1.2/ceph-osd-mgmt.html#remove-an-osd) pointed by @travisn in this NEW orchestrator module. I mean , the` ceph osd drain `command must leave the osd ready to be removed.